homeassistant/components/ads/light.py

```diff
@@ -60,18 +83,37 @@ def __init__(
         ads_hub: AdsHub,
         ads_var_enable: str,
         ads_var_brightness: str | None,
+        ads_var_color_temp_kelvin: str | None,
+        min_color_temp_kelvin: int | None,
+        max_color_temp_kelvin: int | None,
         name: str,
     ) -> None:
         """Initialize AdsLight entity."""
         super().__init__(ads_hub, name, ads_var_enable)
         self._state_dict[STATE_KEY_BRIGHTNESS] = None
+        self._state_dict[STATE_KEY_COLOR_TEMP_KELVIN] = None
         self._ads_var_brightness = ads_var_brightness
+        self._ads_var_color_temp_kelvin = ads_var_color_temp_kelvin
+        self._min_color_temp_kelvin = min_color_temp_kelvin
+        self._max_color_temp_kelvin = max_color_temp_kelvin
+
+        # Determine supported color modes
+        supported_modes = {ColorMode.ONOFF}
         if ads_var_brightness is not None:
-            self._attr_color_mode = ColorMode.BRIGHTNESS
-            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
-        else:
-            self._attr_color_mode = ColorMode.ONOFF
-            self._attr_supported_color_modes = {ColorMode.ONOFF}
+            supported_modes.add(ColorMode.BRIGHTNESS)
+        if ads_var_color_temp_kelvin is not None:
+            supported_modes.add(ColorMode.COLOR_TEMP)
+
+        self._attr_supported_color_modes = supported_modes
+        self._attr_color_mode = next(iter(supported_modes))
```

Comment:
The order of elements in a set can differ from the sequence in which they were added.
Please use a `list` to avoid an unexpected order and then transform it to a `set` for `self._attr_supported_color_modes`.
Or keep it simple with a similar pattern to the one that was being used before the changes, where `self._attr_color_mode` is set and a color mode is added to `self._attr_supported_color_modes` if the condition is meet.